chia/full_node/hint_store.py

```diff
@@ -0,0 +1,39 @@
+from typing import List, Tuple
+import aiosqlite
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.db_wrapper import DBWrapper
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class HintStore:
+    coin_record_db: aiosqlite.Connection
+    db_wrapper: DBWrapper
+
+    @classmethod
+    async def create(cls, db_wrapper: DBWrapper):
+        self = cls()
+        self.db_wrapper = db_wrapper
+        self.coin_record_db = db_wrapper.db
+        # the coin_id is unique, same hint can be used for multiple coins
+        await self.coin_record_db.execute(("CREATE TABLE IF NOT EXISTS hints(coin_id blob PRIMARY KEY,  hint blob)"))
+        await self.coin_record_db.execute("CREATE INDEX IF NOT EXISTS hint_index on hints(hint)")
+        await self.coin_record_db.commit()
+        return self
+
+    async def get_hints(self, hint: bytes) -> List[bytes32]:
```

Comment:
this function is called `get_hints()` but returns coin-IDs. It seems better names would be: `lookup_hints()` or `get_coin_ids_from_hints()`